api/v1alpha1/addon_types.go

```diff
@@ -257,4 +257,28 @@ type ServiceLoadBalancer struct {
 	// +kubebuilder:validation:Enum=MetalLB
 	// +kubebuilder:validation:Required
 	Provider string `json:"provider"`
+
+	// Configuration for the chosen ServiceLoadBalancer provider.
+	// +kubebuilder:validation:Optional
+	Configuration *ServiceLoadBalancerConfiguration `json:"configuration"`
+}
+
+type ServiceLoadBalancerConfiguration struct {
+	// AddressRanges is a list of IPv4 or IPv6 address ranges from which
+	// the provider can choose a load balancer address.
+	AddressRanges []AddressRange `json:"addressRanges"`
```

Comment:
Should these be optional?
```suggestion
	// AddressRanges is a list of IPv4 or IPv6 address ranges from which
	// the provider can choose a load balancer address.
	// +kubebuilder:validation:Optional
	AddressRanges []AddressRange `json:"addressRanges,omitempty"`
```

Reply:
Nope, they're required. Will fix.